src/NuGet.Core/NuGet.Protocol/EnhancedHttpRetryHelper.cs

```diff
@@ -93,6 +113,29 @@ public EnhancedHttpRetryHelper(IEnvironmentVariableReader environmentVariableRea
         /// </summary>
         internal bool Retry429 => _retry429 ??= GetBoolFromEnvironmentVariable(Retry429EnvironmentVariableName, defaultValue: true, _environmentVariableReader);
 
+        /// <summary>
+        /// Gets a value indicating whether or not to observe the Retry-After header on HTTP responses.
+        /// </summary>
+        internal bool ObserveRetryAfter => _observeRetryAfter ??= GetBoolFromEnvironmentVariable(ObserveRetryAfterEnvironmentVariableName, defaultValue: DefaultObserveRetryAfter, _environmentVariableReader);
+
+        internal TimeSpan? MaxRetryAfterDelay
```

Comment:
So seems like we don't have a max default, is that correct? 

Should we have one, even if it's outrageously long?

Reply:
I'm thinking along the same lines of what's done with regex, eve if this is a lot more unpredictable.